sdk/src/test/scala/com/horizen/consensus/ConsensusDataProviderTest.scala

```diff
@@ -0,0 +1,272 @@
+    val blockInfo =
+      new SidechainBlockInfo(0, 0, parentId, timestamp, ModifierSemanticValidity.Valid, Seq(), dummyWithdrawalEpochInfo, vrfProof, vrfProofHash)
+
+    (newBlockId, blockInfo)
+  }
+
+
+  private def slotsPresentationToVrfData(slotsRepresentations: List[List[Int]], prefix: String = ""):  List[List[Option[(VrfProof, VrfProofHash)]]] = {
+    slotsRepresentations.zipWithIndex.map{case (slotsRepresentationsForEpoch, epochIndex) =>
+      slotsRepresentationsForEpoch.zipWithIndex.map{
+        case (1, slotIndex) =>
+          Some(new VrfProof(s"${prefix}proof${epochIndex}${slotIndex}".getBytes), new VrfProofHash(s"${prefix}proofHash${epochIndex}${slotIndex}".getBytes))
+        case (0, _) =>
+          None
+        case _ => throw new IllegalArgumentException
+      }
+    }
+  }
+
+  def getFullConsensusEpochInfoForBlock(idInfo: (ModifierId, SidechainBlockInfo)): FullConsensusEpochInfo = {
+    getFullConsensusEpochInfoForBlock(idInfo._1, idInfo._2)
+  }
+
+  def getInfoForCheckingBlockInEpochNumber(epochNumber: Int): FullConsensusEpochInfo = {
+    getFullConsensusEpochInfoForBlock(blockIdAndInfosPerEpoch(epochNumber - 1).head)
+  }
+}
+
+class BlocksInfoProvider extends SidechainBlockInfoProvider {
+  private val storage = mutable.Map[ModifierId, SidechainBlockInfo]()
+
+  override def blockInfoById(blockId: ModifierId): SidechainBlockInfo = storage(blockId)
+  def addBlockInfo(blockId: ModifierId, sidechainBlockInfo: SidechainBlockInfo): Unit = storage.put(blockId, sidechainBlockInfo)
+}
+
+class ConsensusDataProviderTest extends CompanionsFixture{
+  val generator: SidechainBlockFixture = new SidechainBlockFixture {}
+  val dummyWithdrawalEpochInfo = utils.WithdrawalEpochInfo(0, 0)
+
+
+  @Test
+  def test(): Unit = {
+    val slotsInEpoch = 10
+    val slotsPresentation: List[List[Int]] = List(
+//   1 -- block in slot is present; 0 -- no block for slot
+//   slots 1  2  3  4  5  6  7  8  9  10
+      List(1, 1, 1, 1, 1, 1, 1, 1, 1, 1), //2 epoch
+      List(1, 1, 0, 0, 1, 1, 1, 1, 0, 0), //3 epoch
+      List(1, 1, 0, 0, 1, 1, 1, 1, 0, 0), //4 epoch
+      List(0, 0, 0, 0, 1, 1, 1, 1, 0, 0), //5 epoch
+      List(1, 0, 0, 0, 0, 0, 0, 0, 0, 1), //6 epoch
+      List(1, 0, 1, 0, 1, 0, 1, 0, 1, 0), //7 epoch
+      List(1, 0, 1, 0, 1, 0, 1, 0, 1, 0), //8 epoch
+      List(0, 1, 1, 0, 1, 0, 1, 1, 1, 1), //9 epoch
+      List(0, 1, 1, 0, 1, 0, 1, 1, 0, 1), //10 epoch
+
+    )
+    require(slotsPresentation.forall(_.size == slotsInEpoch))
+
+    val genesisBlockId = bytesToId(Utils.doubleSHA256Hash("genesis".getBytes()))
+    val genesisBlockTimestamp = 1000000
+    val networkParams = TestNetParams(
+      sidechainGenesisBlockId = genesisBlockId,
+      sidechainGenesisBlockTimestamp = genesisBlockTimestamp,
+      consensusSlotsInEpoch = slotsInEpoch,
+      consensusSecondsInSlot = 100
+    )
+
+    val tested1 = new TestedConsensusDataProvider(slotsPresentation, networkParams)
```

Comment:
Please, check the names of the variables in test cases. It's not so clear the context, where they used.
For example:
`tested1` -> `dataProvider1` or `firstDataProvider`
`epochIds1` -> `firstDataProviderEpochIds`.